src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -134,30 +133,36 @@ private async Task PurgePreviousEventVersions()
 
                     foreach (var table in tables)
                     {
-                        var tableRecords = await table.ExecuteQuerySegmentedAsync(new TableQuery<DiagnosticEvent>(), null);
+                        var tableQuery = table.QueryAsync<DiagnosticEvent>(cancellationToken: default);
+                        var tableRecords = new List<DiagnosticEvent>();
+
+                        await foreach (var entity in tableQuery)
```

Comment:
Can we refactor this method to avoid adding to the list? Is there a way to perform all of this in the `await foreach`?